projects/observability/src/pages/explorer/explorer-service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { NavigationParams, NavigationParamsType } from '@hypertrace/common';
+import { Filter, toUrlFilterOperator } from '@hypertrace/components';
+import { ScopeQueryParam } from './explorer.component';
+
+@Injectable({ providedIn: 'root' })
+export class ExplorerService {
+  public buildNavParamsWithFilters(scopeQueryParam: ScopeQueryParam, filters: UrlFilter[]): NavigationParams {
```

Comment:
We're still building the filter by hand here, and overloading what `filter.urlString` means.

If we kept it with the api I mentioned in the other PR, we'd' first go to metadata service to get the attribute, then grab the filter builder that matches that type and used it to build the filter, which should include a prebuilt url string. Something like this (no ide here, might be some issues)

```typescript
    this.metadataServce.getAttribute(scope, key)
       .pipe(
           map(attribute => this.filterBuilderLookupService
            .lookup(attribute.type)
            .buildFilter(attribute, operator, value)),
           map(filter => ({
              navType: NavigationParamsType.InApp,
              path: '/explorer',
              queryParams: {
                filter: filter.urlString,
                scope: scope
              }
          }))
        );
```

Reply:
Done.